### PR TITLE
Divides species_types.dm into different files due to code bloat

### PR DESCRIPTION
**Pull request by Firecage**
This divides the species_types.dm file into several different files, since it was getting quite bloated and annoying to browse through.

### CODE REVIEW

**Comment by XDTM** [comment]
slimepeople should go with jelly, since they're a child of jellypeople

**Comment by Firecage** [comment]
@XDTM Huh, I didn't actually notice that. Time to think of a file name.

**Comment by Kromgar** [comment]
Jellypeople were another option for race change when we had metroids iirc

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+	info_text = "As a <span class='danger'>Titanium Golem</span>, you are immune to ash storms, and slightly more resistant to burn damage."
+	burnmod = 0.9
+
+/datum/species/golem/titanium/on_species_gain(mob/living/carbon/C, datum/species/old_species)
+	. = ..()
+	C.weather_immunities |= "ash"
+
+/datum/species/golem/titanium/on_species_loss(mob/living/carbon/C)
+	. = ..()
+	C.weather_immunities |= "ash"
```
Comment (ChangelingRain) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: &=

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+
+/datum/species/golem/plastitanium/on_species_gain(mob/living/carbon/C, datum/species/old_species)
+	. = ..()
+	C.weather_immunities |= "lava"
+	C.weather_immunities |= "ash"
+
+/datum/species/golem/plastitanium/on_species_loss(mob/living/carbon/C)
+	. = ..()
+	C.weather_immunities |= "ash"
+	C.weather_immunities |= "lava"
```
Comment (ChangelingRain) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: &=

Comment (Firecage) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: Joan? At both of them? Just checking since I didn't write this code.

**Pull request merged**